ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -190,6 +261,43 @@ pub(crate) struct Instruction {
     name: RefCell<Option<String>>,
 }
 
+impl PartialEq for Instruction {
+    fn eq(&self, other: &Self) -> bool {
+        std::ptr::eq(self, other)
+    }
+}
+
+impl Instruction {
+    pub(crate) fn get_operand(&self, index: usize) -> &Operand {
+        self.operands.get(index).unwrap()
+    }
+
+    pub(crate) fn opcode(&self) -> Opcode {
+        self.opcode
+    }
+
+    pub(crate) fn is_store(&self) -> bool {
+        self.opcode == Opcode::Store
+    }
+
+    pub(crate) fn is_gep(&self) -> bool {
+        self.opcode == Opcode::GetElementPtr
+    }
+
+    pub(crate) fn is_control_point(&self, aot_mod: &Module) -> bool {
+        if self.opcode == Opcode::Call {
+            let op = &self.operands[0];
```

Comment:
I guess `Opcode::Call` always has (at least?) one operand so this can't panic?

Reply:
the first operand would be a callee, so yeah.

Reply:
Documented in c54d023c1e017ee88e4cb050bc0b6c0d256d2d47